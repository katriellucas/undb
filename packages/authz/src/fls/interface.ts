import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import type { FLS } from './fls.js'
import type { WithFLSFieldId } from './specifications/fls-field-id.specification.js'
import type { WithFLSId } from './specifications/fls-id.specification.js'
import type {
  WithFLSAction,
  WithFLSActionIn,
  WithFLSPolicy,
  WithFLSPolicyFilter,
} from './specifications/fls-policy.specification.js'
import type { FLSSubjectContainsUser, WithFLSSubjects } from './specifications/fls-subject.specification.js'
import type { WithFLSTableId } from './specifications/fls-table-id.specification.js'

export interface IFLSVisitor extends ISpecVisitor {
  withId(s: WithFLSId): void
  withFLSPolicy(s: WithFLSPolicy): void
  withFLSPolicyAction(s: WithFLSAction): void
  actionsIn(s: WithFLSActionIn): void
  withFLSPolicyFilter(s: WithFLSPolicyFilter): void
  withTableId(s: WithFLSTableId): void
  withFieldId(s: WithFLSFieldId): void
  withFLSSubjects(s: WithFLSSubjects): void
  subjectContainsUser(s: FLSSubjectContainsUser): void
}

export type FLSSpecification = CompositeSpecification<FLS, IFLSVisitor>
