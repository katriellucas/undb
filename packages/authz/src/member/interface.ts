import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import type { Member } from './member.js'
import type { WithMemberId } from './specifications/member-id.specification.js'
import type { WithMemberRole } from './specifications/member-role.specification.js'
import type { WithMemberUserLike } from './specifications/member-search.specification.js'
import type { WithMemberUserId } from './specifications/member-user-id.specification.js'

export interface IMemberVisitor extends ISpecVisitor {
  withMemberId(s: WithMemberId): void
  withMemberUserId(s: WithMemberUserId): void
  withMemberRole(s: WithMemberRole): void
  userLike(s: WithMemberUserLike): void
}

export type MemberSpecification = CompositeSpecification<Member, IMemberVisitor>
