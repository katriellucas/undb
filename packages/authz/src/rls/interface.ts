import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import type { RLS } from './rls.js'
import type { WithRLSId } from './specifications/rls-id.specification.js'
import type {
  WithRLSAction,
  WithRLSActionIn,
  WithRLSPolicy,
  WithRLSPolicyFilter,
} from './specifications/rls-policy.specification.js'
import type { RLSSubjectContainsUser, WithRLSSubjects } from './specifications/rls-subject.specification.js'
import type { WithRLSTableId } from './specifications/rls-table-id.specification.js'

export interface IRLSVisitor extends ISpecVisitor {
  withId(s: WithRLSId): void
  withTableId(s: WithRLSTableId): void
  withRLSPolicy(s: WithRLSPolicy): void
  withRLSPolicyAction(s: WithRLSAction): void
  withRLSPolicyFilter(s: WithRLSPolicyFilter): void
  withRLSSubjects(s: WithRLSSubjects): void
  actionsIn(s: WithRLSActionIn): void
  subjectContainsUser(s: RLSSubjectContainsUser): void
}

export type RLSSpecification = CompositeSpecification<RLS, IRLSVisitor>
