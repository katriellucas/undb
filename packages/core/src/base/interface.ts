import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import type { Base } from './base.js'
import type { WithBaseId } from './specifications/base-id.specification.js'
import type { WithBaseName } from './specifications/base-name.specification.js'
import type { WithBaseQ } from './specifications/base-q.specification.js'

export interface IBaseSpecVisitor extends ISpecVisitor {
  withId(v: WithBaseId): void
  withName(v: WithBaseName): void
  withQ(v: WithBaseQ): void
}

export type BaseSpecification = CompositeSpecification<Base, IBaseSpecVisitor>
