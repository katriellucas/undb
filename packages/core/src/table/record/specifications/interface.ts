import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import { type ISpecification } from '@undb/domain'
import { type Record } from '../record.js'
import type { HasExtension, HasFileType, IsAttachmentEmpty } from './attachment.specification.js'
import type { BoolIsFalse, BoolIsTrue } from './bool.specification.js'
import type { CollaboratorEqual, CollaboratorIsEmpty } from './collaborator.specification.js'
import type {
  DateRangeDateEqual,
  DateRangeDateGreaterThan,
  DateRangeDateGreaterThanOrEqual,
  DateRangeDateLessThan,
  DateRangeDateLessThanOrEqual,
  DateRangeEmpty,
  DateRangeEqual,
} from './date-range.specification.js'
import type {
  DateBetween,
  DateEqual,
  DateGreaterThan,
  DateGreaterThanOrEqual,
  DateIsToday,
  DateIsTomorrow,
  DateIsYesterday,
  DateLessThan,
  DateLessThanOrEqual,
} from './date.specification.js'
import type { JsonEmpty } from './json.specification.js'
import type { MultiSelectEqual, MultiSelectIn, MultiSelectIsEmpty } from './multi-select.specification.js'
import type {
  NumberEmpty,
  NumberEqual,
  NumberGreaterThan,
  NumberGreaterThanOrEqual,
  NumberLessThan,
  NumberLessThanOrEqual,
} from './number.specification.js'
import type { ParentAvailableSpec } from './parent.specification.js'
import type { WithRecordAutoIncrement } from './record-auto-increment.specification.js'
import type { WithRecordCreatedAt } from './record-created-at.specification.js'
import type { CreatedByIn, WithRecordCreatedBy } from './record-created-by.specification.js'
import type { WithRecordId, WithRecordIds } from './record-id.specification.js'
import type { WithRecordLike } from './record-search.specification.js'
import type { WithRecordTableId } from './record-table-id.specification.js'
import type { WithRecordUpdatedAt } from './record-updated-at.specification.js'
import type { UdpatedByIn, WithRecordUpdatedBy } from './record-updated-by.specification.js'
import type { WithRecordValues } from './record-values.specification.js'
import type { ReferenceEqual } from './reference.specification.js'
import type { SelectEmpty, SelectEqual, SelectIn } from './select.specification.js'
import type {
  StringContain,
  StringEndsWith,
  StringEqual,
  StringRegex,
  StringStartsWith,
} from './string.specification.js'
import type { IsTreeRoot, TreeAvailableSpec } from './tree.specification.js'

interface IRecordSpecVisitor {
  idEqual(s: WithRecordId): void
  idsIn(s: WithRecordIds): void
  tableIdEqual(s: WithRecordTableId): void

  createdAt(s: WithRecordCreatedAt): void
  createdBy(s: WithRecordCreatedBy): void
  createdByIn(s: CreatedByIn): void
  updatedAt(s: WithRecordUpdatedAt): void
  updatedBy(s: WithRecordUpdatedBy): void
  updatedByIn(s: UdpatedByIn): void

  autoIncrement(s: WithRecordAutoIncrement): void

  values(s: WithRecordValues): void
}

interface IRecordValueVisitor {
  stringEqual(s: StringEqual): void
  stringContain(s: StringContain): void
  stringStartsWith(s: StringStartsWith): void
  stringEndsWith(s: StringEndsWith): void
  stringRegex(s: StringRegex): void
  stringEmpty(s: StringEqual): void

  like(s: WithRecordLike): void

  numberEqual(s: NumberEqual): void
  numberGreaterThan(s: NumberGreaterThan): void
  numberLessThan(s: NumberLessThan): void
  numberGreaterThanOrEqual(s: NumberGreaterThanOrEqual): void
  numberLessThanOrEqual(s: NumberLessThanOrEqual): void
  numberEmpty(s: NumberEmpty): void

  dateEqual(s: DateEqual): void
  dateGreaterThan(s: DateGreaterThan): void
  dateLessThan(s: DateLessThan): void
  dateGreaterThanOrEqual(s: DateGreaterThanOrEqual): void
  dateLessThanOrEqual(s: DateLessThanOrEqual): void
  dateIsToday(s: DateIsToday): void
  dateIsTomorrow(s: DateIsTomorrow): void
  dateIsYesterday(s: DateIsYesterday): void
  dateBetween(s: DateBetween): void

  dateRangeEqual(s: DateRangeEqual): void
  dateRangeEmpty(s: DateRangeEmpty): void
  dateRangeDateEqual(s: DateRangeDateEqual): void
  dateRangeDateGreaterThan(s: DateRangeDateGreaterThan): void
  dateRangeDateLessThan(s: DateRangeDateLessThan): void
  dateRangeDateGreaterThanOrEqual(s: DateRangeDateGreaterThanOrEqual): void
  dateRangeDateLessThanOrEqual(s: DateRangeDateLessThanOrEqual): void

  collaboratorEqual(s: CollaboratorEqual): void
  collaboratorIsEmpqy(s: CollaboratorIsEmpty): void

  multiSelectEqual(s: MultiSelectEqual): void
  multiSelectIn(s: MultiSelectIn): void
  multiSelectIsEmpty(s: MultiSelectIsEmpty): void

  selectEqual(s: SelectEqual): void
  selectIn(s: SelectIn): void
  selectEmpty(s: SelectEmpty): void

  boolIsTrue(s: BoolIsTrue): void
  boolIsFalse(s: BoolIsFalse): void

  referenceEqual(s: ReferenceEqual): void

  treeAvailable(s: TreeAvailableSpec): void
  isTreeRoot(s: IsTreeRoot): void

  parentAvailable(s: ParentAvailableSpec): void

  jsonEmpty(s: JsonEmpty): void

  hasFileType(s: HasFileType): void
  hasExtension(s: HasExtension): void
  isAttachmentEmpty(s: IsAttachmentEmpty): void
}

export type RecordCompositeSpecification = CompositeSpecification<Record, IRecordVisitor>

export type IRecordSpec = ISpecification<Record, IRecordVisitor>

export type IRecordVisitor = IRecordSpecVisitor & IRecordValueVisitor & ISpecVisitor
