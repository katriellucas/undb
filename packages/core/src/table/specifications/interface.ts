import type { CompositeSpecification, ISpecVisitor } from '@undb/domain'
import { type ISpecification } from '@undb/domain'
import type { WithAggregateFieldId } from '../field/index.js'
import type {
  WithFieldDescription,
  WithFieldDisplay,
  WithFieldName,
} from '../field/specifications/base-field.specification.js'
import type { WithCurrencySymbol } from '../field/specifications/currency-field.specification.js'
import type { WithFormat, WithTimeFormat } from '../field/specifications/date-field.specification.js'
import type { WithFieldRequirement } from '../field/specifications/field-constraints.specification.js'
import type {
  WithDuplicatedField,
  WithNewFieldType,
  WithoutField,
} from '../field/specifications/field.specification.js'
import type { WithReferenceFieldId } from '../field/specifications/lookup-field.specification.js'
import type { WithRatingMax } from '../field/specifications/rating-field.specification.js'
import type {
  WithDisplayFields,
  WithForeignTableId,
  WithSymmetricReferenceField,
} from '../field/specifications/reference-field.specification.js'
import type {
  WithNewOption,
  WithOption,
  WithOptions,
  WithoutOption,
} from '../field/specifications/select-field.specification.js'
import type {
  WithFormFieldFilter,
  WithFormFieldsRequirements,
  WithFormFieldsSpecification,
  WithFormFieldsVisibility,
} from '../form/index.js'
import type { WithFormFieldsOrder } from '../form/specifications/form-fields-order.specification.js'
import type { WithFormName } from '../form/specifications/form-name.specification.js'
import type { WithNewForm, WithTableForms, WithoutForm } from '../form/specifications/form.specification.js'
import { type Table } from '../table.js'
import type {
  WithWidgetSpecification,
  WithWidgetsLayout,
  WithoutWidgetSpecification,
} from '../view/dashboard/specifications/widget.specification.js'
import type { WithGalleryField, WithGanttField, WithKanbanField, WithViewsOrder } from '../view/index.js'
import type { WithCalendarField } from '../view/specifications/calendar.specification.js'
import type { WithDisplayType } from '../view/specifications/display-type.specification.js'
import type { WithFilter } from '../view/specifications/filters.specificaiton.js'
import type { WithRowHeight } from '../view/specifications/row-height.specification.js'
import type { WithShowSystemFieldsSpec } from '../view/specifications/show-system-fields.specification.js'
import type { WithSorts } from '../view/specifications/sorts.specification.js'
import type { WithTreeViewField } from '../view/specifications/tree-view.specification.js'
import type {
  WithFieldOption,
  WithFieldVisibility,
  WithFieldWidth,
} from '../view/specifications/view-field-option.specification.js'
import type { WithViewFieldsOrder } from '../view/specifications/view-fields-order.specification.js'
import type { WithViewPinnedFields } from '../view/specifications/view-pinned-fields.specification.js'
import type {
  WithNewView,
  WithTableView,
  WithTableViews,
  WithViewName,
  WithoutView,
} from '../view/specifications/views.specification.js'
import type { WithChartAggregateSpec } from '../visualization/specifications/chart-visualization.specification.js'
import type { WithNumberAggregateSpec } from '../visualization/specifications/number-visualization.specification.js'
import type { WithVisualizationFieldSpec } from '../visualization/specifications/visualization-field.specification.js'
import type { WithVisualizationNameSpec } from '../visualization/specifications/visualization-name.specification.js'
import type { WithTableBaseId } from './table-base-id.specification.js'
import type { WithTableEmoji } from './table-emoji.specification.js'
import type { WithNewField } from './table-field.specification.js'
import type { WithTableId, WithTableIds } from './table-id.specification.js'
import type { WithTableName } from './table-name.specification.js'
import type { WithTableSchema } from './table-schema.specification.js'
import type { WithTableFormId, WithTableViewId } from './table-view-id.specification.js'

export interface ITableSpecVisitor extends ISpecVisitor {
  idEqual(s: WithTableId): void
  idsIn(s: WithTableIds): void
  nameEqual(s: WithTableName): void
  emojiEqual(s: WithTableEmoji): void
  schemaEqual(s: WithTableSchema): void
  viewIdEqual(s: WithTableViewId): void
  viewsEqual(s: WithTableViews): void
  viewEqual(s: WithTableView): void
  viewNameEqual(s: WithViewName): void
  newView(s: WithNewView): void
  withoutView(s: WithoutView): void
  viewsOrderEqual(s: WithViewsOrder): void

  baseIdEq(s: WithTableBaseId): void

  formsEqual(s: WithTableForms): void
  withFormName(s: WithFormName): void
  formFieldsEqual(s: WithFormFieldsSpecification): void
  formFieldsOrder(s: WithFormFieldsOrder): void
  withFormFieldsVisibility(s: WithFormFieldsVisibility): void
  withFormFieldsRequirements(s: WithFormFieldsRequirements): void
  withFormFieldFilter(s: WithFormFieldFilter): void
  formIdEqual(s: WithTableFormId): void
  newForm(s: WithNewForm): void
  withoutForm(s: WithoutForm): void

  sortsEqual(s: WithSorts): void

  filterEqual(s: WithFilter): void
  fieldsOrder(s: WithViewFieldsOrder): void
  fieldOptionsEqual(s: WithFieldOption): void
  fieldWidthEqual(s: WithFieldWidth): void
  fieldVisibility(s: WithFieldVisibility): void
  pinnedFields(s: WithViewPinnedFields): void

  ratingMaxEqual(s: WithRatingMax): void

  currencySymbolEqual(s: WithCurrencySymbol): void

  rowHeightEqual(s: WithRowHeight): void
  displayTypeEqual(s: WithDisplayType): void
  kanbanFieldEqual(s: WithKanbanField): void
  galleryFieldEqual(s: WithGalleryField): void
  ganttFieldEqual(s: WithGanttField): void
  calendarFieldEqual(s: WithCalendarField): void
  treeViewFieldEqual(s: WithTreeViewField): void

  newField(s: WithNewField): void
  withoutField(s: WithoutField): void
  withDuplicatedField(s: WithDuplicatedField): void

  withNewFieldType(s: WithNewFieldType): void

  optionsEqual(s: WithOptions): void
  optionEqual(s: WithOption): void
  newOption(s: WithNewOption): void
  withoutOption(s: WithoutOption): void

  withFieldName(s: WithFieldName): void
  withFieldDescription(s: WithFieldDescription): void
  withFieldDisplay(s: WithFieldDisplay): void
  displayFieldsEqual(s: WithDisplayFields): void
  withFormat(s: WithFormat): void
  withTimeFormat(s: WithTimeFormat): void

  withShowSystemFields(s: WithShowSystemFieldsSpec): void

  withFieldRequirement(s: WithFieldRequirement): void
  symmetricReferenceFieldEqual(s: WithSymmetricReferenceField): void
  foreignTableIdEqual(s: WithForeignTableId): void

  withWidget(s: WithWidgetSpecification): void
  withoutWidget(s: WithoutWidgetSpecification): void
  withWidgetsLayout(s: WithWidgetsLayout): void
  withVisualizationName(s: WithVisualizationNameSpec): void
  withVisualizationField(s: WithVisualizationFieldSpec): void
  withNumberAggregate(s: WithNumberAggregateSpec): void
  withChartAggregate(s: WithChartAggregateSpec): void

  withReferenceFieldId(s: WithReferenceFieldId): void
  withAggregateFieldId(s: WithAggregateFieldId): void
}

export type ITableSpec = ISpecification<Table, ITableSpecVisitor>

export type TableCompositeSpecification = CompositeSpecification<Table, ITableSpecVisitor>
