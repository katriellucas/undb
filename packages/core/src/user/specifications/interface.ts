import type { CompositeSpecification } from '@undb/domain'
import type { User } from '../user.js'
import type { WithUserAvatar } from './user-avatar.specification.js'
import type { WithUserColor } from './user-color.js'
import type { WithUserEmail } from './user-email.specification.js'
import type { WithUserId, WithUserIds } from './user-id.specification.js'
import type { WithUsername } from './username.specification.js'

export interface IUserSpecVisitor {
  idEqual(s: WithUserId): void
  idsIn(s: WithUserIds): void
  avatarEqual(s: WithUserAvatar): void
  emailEqual(s: WithUserEmail): void
  usernameEqual(s: WithUsername): void
  colorEqual(s: WithUserColor): void
  or(left: UserSpecification, right: UserSpecification): IUserSpecVisitor
  not(): IUserSpecVisitor
}

export type UserSpecification = CompositeSpecification<User, IUserSpecVisitor>
