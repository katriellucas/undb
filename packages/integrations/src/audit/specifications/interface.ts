import type { CompositeSpecification } from '@undb/domain'
import type { Audit } from '../audit.js'
import type { WithAuditDetail } from './audit-detail.specification.js'
import type { WithAuditId } from './audit-id.specification.js'
import type { WithAuditOp } from './audit-op.specification.js'
import type { WithAuditOperator } from './audit-operaotr.specification.js'
import type { WithAuditTableId } from './audit-table-id.specification.js'
import type { WithAuditTarget } from './audit-target.specification.js'
import type { WithAuditAfter, WithAuditTimestamp } from './audit-timestamp.specification.js'

export interface IAuditSpecVisitor {
  idEqual(s: WithAuditId): void
  timestampEqual(s: WithAuditTimestamp): void
  op(s: WithAuditOp): void
  operatorEqual(s: WithAuditOperator): void
  targetEqual(s: WithAuditTarget): void
  detailEqual(s: WithAuditDetail): void
  tableIdEqual(s: WithAuditTableId): void
  after(s: WithAuditAfter): void

  or(left: AuditSpecification, right: AuditSpecification): IAuditSpecVisitor
  not(): IAuditSpecVisitor
}

export type AuditSpecification = CompositeSpecification<Audit, IAuditSpecVisitor>
