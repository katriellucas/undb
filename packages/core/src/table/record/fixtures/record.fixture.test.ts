import { NumberFieldValue, StringFieldValue } from '../../field/index.js'
import { NumberEqual, StringEqual } from '../specifications/index.js'
import { RecordCompositeSpecification } from '../specifications/interface.js'
import { createTestRecord } from './record.fixture.js'

beforeAll(() => {
  vi.setSystemTime(new Date(2022, 1, 1))
})

test.each<RecordCompositeSpecification[][]>([
  [[new StringEqual('hello', new StringFieldValue('world'))]],
  [[new NumberEqual('hello', new NumberFieldValue(1))]],
  [[new StringEqual('hello', new StringFieldValue('world')), new NumberEqual('number', new NumberFieldValue(1))]],
])('createTestRecord', (specs) => {
  const record = createTestRecord(...specs)
  expect(record).toMatchSnapshot()
})
