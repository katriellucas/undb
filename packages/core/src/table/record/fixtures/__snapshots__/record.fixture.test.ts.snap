// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createTestRecord 1`] = `
Record {
  "autoIncrement": undefined,
  "createdAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "createdBy": undefined,
  "createdByProfile": null,
  "displayValues": RecordDisplayValues {
    "props": {},
  },
  "id": RecordId {
    "props": {
      "value": "record",
    },
  },
  "tableId": TableId {
    "props": {
      "value": "tableId",
    },
  },
  "updatedAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "updatedBy": undefined,
  "updatedByProfile": null,
  "values": RecordValues {
    "props": Map {
      "hello" => StringFieldValue {
        "props": {
          "value": "world",
        },
      },
    },
  },
}
`;

exports[`createTestRecord 2`] = `
Record {
  "autoIncrement": undefined,
  "createdAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "createdBy": undefined,
  "createdByProfile": null,
  "displayValues": RecordDisplayValues {
    "props": {},
  },
  "id": RecordId {
    "props": {
      "value": "record",
    },
  },
  "tableId": TableId {
    "props": {
      "value": "tableId",
    },
  },
  "updatedAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "updatedBy": undefined,
  "updatedByProfile": null,
  "values": RecordValues {
    "props": Map {
      "hello" => NumberFieldValue {
        "props": {
          "value": 1,
        },
      },
    },
  },
}
`;

exports[`createTestRecord 3`] = `
Record {
  "autoIncrement": undefined,
  "createdAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "createdBy": undefined,
  "createdByProfile": null,
  "displayValues": RecordDisplayValues {
    "props": {},
  },
  "id": RecordId {
    "props": {
      "value": "record",
    },
  },
  "tableId": TableId {
    "props": {
      "value": "tableId",
    },
  },
  "updatedAt": r {
    "props": {
      "value": 2022-02-01T00:00:00.000Z,
    },
  },
  "updatedBy": undefined,
  "updatedByProfile": null,
  "values": RecordValues {
    "props": Map {
      "hello" => StringFieldValue {
        "props": {
          "value": "world",
        },
      },
      "number" => NumberFieldValue {
        "props": {
          "value": 1,
        },
      },
    },
  },
}
`;
